File: c#/P2.cs
public class P2
{
  private static int Fibonacci(int a, int b, int n)
  {
    if(n == 0)
    {
      return a;
    }

    return Fibonacci(b, a + b, n - 1);
  }

  public static void Main()
  {
    int i = 0;
    int fib = 0;
    long sum = 0L;

    while(fib < 4000000)
    {
      fib = Fibonacci(0, 1, i);
      if(fib % 2 == 0)
      {
        sum += (long) fib;
      }
      i++;
    }

    System.Console.WriteLine(sum);
  }
}

File: php/p2.php
<?php 
function fibonacci($a, $b, $n){
  if($n == 0){
    return $a;
  }
  return fibonacci($b, $a+$b, $n - 1);
}

$i = 0;
$fib = 0;
$sum = 0;
while($fib < 4000000){
  $fib = fibonacci(0, 1, $i);
  if($fib % 2 == 0){
    $sum += $fib;
  }
  $i++;
}
echo $sum."\n";
?>